### PR TITLE
[backport] Fix `no_std` MSRV

### PR DESCRIPTION
**Pull request by apoelstra**
Backport of #690 to 0.28

### CODE REVIEW

**Comment by apoelstra** [comment]
Oops, it looks like #985 beat me to the punch ... @tcharding should we close this one or yours? I believe they are identical except for the name of the branch that we're merging into.

Why is yours marked DO NOT MERGE?

**Comment by tcharding** [comment]
> Why is yours marked DO NOT MERGE?

I wrote that when the target branch was `master` before I realised I could target `0.28.x`.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 target
 Cargo.lock
+dep_test
```
[the remark this answers is not in the document]
Comment (tcharding) [reply .gitignore:3]: I don't get why you mention the test crate. This directory is created in the `test.sh` script. This chage, AFAICT, helps devs running the test script with `AS_DEPENDENCY` set to true.

**Review by tcharding — APPROVED** [review]
ACK e45f6c720c3e6a97ec54d993ecdfc7b5fa0810e4

**Review by mcroad — APPROVED** [review]

**Comment by apoelstra** [comment]
Merged, tagged and published.

**Comment by devrandom** [comment]
used in https://github.com/lightningdevkit/rust-lightning/pull/1389

**Pull request merged**